feat(variables): added `arrayFuzzySearch` variable (#3002)

Oceanity opened the pull request:
### Description of the Change
Introduced an `arrayFuzzySearch` variable that allows a user to find the closest match in an array for a provided search


### Applicable Issues
#3002 


### Testing
Tested all 3 provided examples to ensure they work as expected, and also tested passing in a simple custom variable array and an array from an HTTP Request response


### Screenshots
n/a


<!--
Note:
  Please be aware that we may require changes if we 
  believe they are needed to meet the vision and standards of Firebot.
  Don't take suggestions for tweaks personally, we are all simply trying to make Firebot
  the best that it can be :)
-->

phroggster commented:
Thanks Oceanity, I love this! If only everything in computer sciences were fuzzier by default, we'd all be better off!

Two (non-blocking) ideas:
* An optional `minimumSimilarity` argument. Where, if specified, the variable evaluation would return the default value ("null", or see below) should the match with the smallest distance be greater than the argument's numeric-ish value.
* An optional `defaultValue` argument, in similar style to `$userMetadata[...]` or `$customVariable[...]`, so that we can confidently match viewer-supplied text against an (admittedly esoteric) array consisting of, e.g. `["null", "mull", "bull"]`.

To future reviewer(s) of this:
Fuse.js released version 7.1.0 ~2 weeks ago. It adds the ability to optionally ignore diacritic (the fancy little "hats" that some letters can wear in many languages outside of English) character differences. IMHO, the fuse.js dep *should* get bumped to 7.1.0 to support this feature, and we should *all* embrace the fuzzy.

Oceanity commented:
> Thanks Oceanity, I love this! If only everything in computer sciences were fuzzier by default, we'd all be better off!
> 
> Two (non-blocking) ideas:
> 
>     * An optional `minimumSimilarity` argument. Where, if specified, the variable evaluation would return the default value ("null", or see below) should the match with the smallest distance be greater than the argument's numeric-ish value.
> 
>     * An optional `defaultValue` argument, in similar style to `$userMetadata[...]` or `$customVariable[...]`, so that we can confidently match viewer-supplied text against an (admittedly esoteric) array consisting of, e.g. `["null", "mull", "bull"]`.
> 
> 
> To future reviewer(s) of this: Fuse.js released version 7.1.0 ~2 weeks ago. It adds the ability to optionally ignore diacritic (the fancy little "hats" that some letters can wear in many languages outside of English) character differences. IMHO, the fuse.js dep _should_ get bumped to 7.1.0 to support this feature, and we should _all_ embrace the fuzzy.

Added in [da8c8fa](https://github.com/crowbartools/Firebot/pull/3003/commits/da8c8fa3fe63b2e9ece45d8155da4a09ecdd71a8)

ebiggz reviewed (commented):
Thanks for working on this @Oceanity! 

I agree with @phroggster that it would be worth updating fuse.js to latest. We are already on 7.x.x and from the change log I am not seeing any notable breaking changes. If you could bump fuses version in this PR, that would be great. Then we can get this merged in 👌🏻

Oceanity commented:
> Thanks for working on this @Oceanity!
> 
> I agree with @phroggster that it would be worth updating fuse.js to latest. We are already on 7.x.x and from the change log I am not seeing any notable breaking changes. If you could bump fuses version in this PR, that would be great. Then we can get this merged in 👌🏻

Updated Fuse.js to latest ([d31ad98](https://github.com/crowbartools/Firebot/pull/3003/commits/d31ad981ba9e98aae6e616f40f46889f17e4e2fd)) and added param to ignore Diacritics ([8ee0ba2](https://github.com/crowbartools/Firebot/pull/3003/commits/8ee0ba2c2c6b0693d14e65f2c79305c43aed33fa))!
